x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -314,6 +323,36 @@ public static Query doTranslate(Range r, TranslatorHandler handler) {
         }
     }
 
+    public static class InComparisons extends ExpressionTranslator<In> {
+
+        protected Query asQuery(In in, TranslatorHandler handler) {
+            return doTranslate(in, handler);
+        }
+
+        public static Query doTranslate(In in, TranslatorHandler handler) {
+            Query q;
+            if (in.value() instanceof FieldAttribute) {
+                // equality should always be against an exact match (which is important for strings)
+                FieldAttribute fa = (FieldAttribute) in.value();
+                List<Expression> list = in.list();
+
+                // TODO: this needs to be handled inside the optimizer
+                list.removeIf(e -> DataTypes.isNull(e.dataType()));
+                DataType dt = list.get(0).dataType();
+                Set<Object> set = new LinkedHashSet<>(CollectionUtils.mapSize(list.size()));
+
+                for (Expression e : list) {
+                    set.add(valueOf(e));
```

Comment:
Behavior slightly changed. See comment in SQL
https://github.com/elastic/elasticsearch/pull/53244/files#r394547453